content/03-reference/01-tools-and-interfaces/02-prisma-client/090-raw-database-access.mdx

```diff
@@ -26,71 +44,205 @@ const result = await prisma.queryRaw('SELECT * FROM User;')
 
-While `executeRaw` always returns `Promise<number>`, the return type of `queryRaw` can be adjusted with a TypeScript Generic.
-It has the following function signature:
+As an alternative to tagged templates, `queryRaw()` and `executeRaw()` support standard parameterized queries where each variable is represented by a symbol (`?` for mySQL, `$1`, `$2`, and so on for PostgreSQL):
 
 ```ts
-queryRaw<T = any>(query: string | TemplateStringsArray): Promise<T>;
+const userName = "Sarah";
+const email = "alice@prisma.io";
+const result = await prisma.queryRaw(
+  "SELECT * FROM User WHERE (name = ? OR email = ?)", // MySQL variable, represented by ?
+  userName,
+  email
+);
 ```
 
-The return type of `queryRaw` is a `Promise` for the [generic](https://www.typescriptlang.org/docs/handbook/generics.html) type parameter `T`. This means you can type the result manually by providing `T` when you invoke `queryRaw`. If you don't provide any type, the return type of `queryRaw` defaults to `any`.
+As with tagged templates, Prisma Client escapes all variables. 
+
+> **Note**: You cannot pass a table or column name as a variable into a parameterized query.
+
+## SQL injection
+
+Prisma Client mitigates the risk of SQL injection in the following ways:
+
+* Prisma Client escapes all variables when you use tagged templates (``` queryRaw`` ``` or ``` executeRaw`` ``` ) or parameterized queries (`queryRaw(query, variables)` or `executeRaw(query, variables)`), and sends all queries as prepared statements.
+* Both versions of `executeRaw` (standard and tagged template) can only run **one** query at a time. You cannot append a second query - for example, adding `DROP bobby_tables` to the end of an `ALTER`.
+
+If you cannot use tagged templates or parameterized queries (for example, you need to pass in table or column names as variables), be aware that your code may be vulnerable to SQL injection.
+
+### Warning: String concatenation 
+
+The following example concatanates `query`and `inputString`. Prisma Client ❌ **cannot** escape `inputString` in this example, which makes it vulnerable to SQL injection:
 
 ```ts
-// import the generated `User` type from the `@prisma/client` module
-import { User } from '@prisma/client'
+const inputString = '"Sarah" UNION SELECT id, title, content FROM Post'; // SQL Injection
+const query = "SELECT id, name, email FROM User WHERE name = " + inputString;
+const result = await prisma.queryRaw(query);
 
-const result = await prisma.raw<User[]>('SELECT * FROM User;')
-// result is of type: `User[]`
+console.log(result);
 ```
 
-Now, `result` is statically typed to the generated `User` type (or rather an array thereof) from Prisma Client.
+### Warning: Template literals 
 
-![](https://imgur.com/H2TCRc5.png)
+The following example uses a template literal inside `queryRaw()`and returns all users where `name` equals `${inputString}`. Prisma Client ❌ **cannot** escape `inputString` in this example, which makes it vulnerable to SQL injection:
```

Comment:
Sames here with SQL injection. I now really wonder if I'm not expecting something wrong? I'd generally pluralize that or add `an` in front.